### ISSUE TITLE
[v2] TargetGroupBinding reconciler should not try to create duplicate SG rules

### ISSUE THREAD

**Issue by Pryz** [kubernetes-sigs/aws-alb-ingress-controller#1438]
When managing the rules of a security group, the TargetGroupBinding reconciler should not try to create duplicates. Right now, if we create multiple TG bindings with something like : 

```
apiVersion: elbv2.k8s.aws/v1alpha1
kind: TargetGroupBinding
metadata:
  name: ...
  namespace: ...
spec:
  networking:
    ingress:
       - from:
         - ipBlock:
             cidr: 0.0.0.0/0
    - ports:
       - port: 8080
         protol: tcp
  serviceRef:
    ... 
```

If the ingress rule already exists on another TargetGroupBiding, the following error will be raised : 
```
{"level":"error","ts":1600749371.0291567,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"targetgroupbinding","name":"foo","namespace":"bar","error":"InvalidParameterValue: The same permission must not appear multiple times\n\tstatus code: 400, request id: 610e5c6e-e1a3-4888-a805-4f3bbb783fe5","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:235\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:209\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:188\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/
wait.go:90"} 
```

**Comment by M00nF1sh** [comment]
@Pryz 
Hi, i haven't been able to reproduce this. i have used yaml like 
```
apiVersion: v1
kind: Service
metadata:
  namespace: test-ns
  name: game-2048
spec:
  ports:
    - port: 80
      targetPort: 80
      protocol: TCP
  type: NodePort
  selector:
    app: game-2048
---
apiVersion: elbv2.k8s.aws/v1alpha1
kind: TargetGroupBinding
metadata:
  namespace: test-ns
  name: my-tg-1
spec:
  targetGroupARN: arn:aws:elasticloadbalancing:us-west-2:283511030707:targetgroup/my-tg-1/38d4dadf64a7c98e
  targetType: instance
  serviceRef:
    name: game-2048
    port: 80
  networking:
    ingress:
      - from:
        - ipBlock:
            cidr: 0.0.0.0/0
        ports:
        - port: 8080
          protocol: TCP
---
apiVersion: elbv2.k8s.aws/v1alpha1
kind: TargetGroupBinding
metadata:
  namespace: test-ns
  name: my-tg-2
spec:
  targetGroupARN: arn:aws:elasticloadbalancing:us-west-2:283511030707:targetgroup/my-tg-2/f78e16c2d90a9ce9
  targetType: instance
  serviceRef:
    name: game-2048
    port: 80
  networking:
    ingress:
      - from:
        - ipBlock:
            cidr: 0.0.0.0/0
        - ipBlock:
            cidr: 192.168.0.0/16
        ports:
        - port: 8080
          protocol: TCP
```

**Comment by M00nF1sh** [comment]
Did you manually modified the securityGroup rules on worker node? the controller caches securityGroups rules for 5 minute to improve performance.
If so, we can improve this by parse the error message and fetch updated securityGroup rules

**Comment by Pryz** [comment]
After a debugging session, we found out the root cause : the issue was that a running NLB was still using the built-it k8s controller. For this to work, we set the same rule (9090 opened on 0.0.0.0/0) via an external process (here Terraform).

**Comment by M00nF1sh** [comment]
resolved in v2.0.0-rc2 release. closing this :D
